thetis/shallowwater_eq.py

```diff
@@ -265,10 +267,30 @@ def get_bnd_functions(self, eta_in, uv_in, bnd_id, bnd_conditions):
             h_ext = self.depth.get_total_depth(eta_ext)
             area = h_ext*bnd_len  # NOTE using internal elevation
             uv_ext = funcs['flux']/area*self.normal
-        else:
-            raise Exception('Unsupported bnd type: {:}'.format(funcs.keys()))
+        if eta_ext is None or uv_ext is None:
+            raise Exception('Unsupported open bnd type: {:}'.format(funcs.keys()))
         return eta_ext, uv_ext
 
+    def bnd_state_defined(self, bnd_funcs, bnd_id):
```

Comment:
This is getting a little opaque tbh. Can we add some comments in the doc string that this implies we assume an open boundary, and otherwise it's closed and that get_bnd_functions() should only be called on this bnd_id in the first case.

Reply:
Yes, added a clarifying docstring and changed the name to hopefully more clear `impose_dynamic_bnd`.

Reply:
Ah yeah, I only just saw it was already renamed from `is_open` previously. Clear now with docstring